src/overlays/actors/ovl_En_Dt/z_en_dt.c

```diff
@@ -28,59 +95,714 @@ ActorInit En_Dt_InitVars = {
+            npcMuto = (EnMuto*)this->npcEnMuto;
+            npcBaisen = (EnBaisen*)this->npcEnBaisen;
+        }
+
+        this->showedCouplesMask = false;
+        this->textIdIndex = 0;
+
+        if (Player_GetMask(play) == PLAYER_MASK_COUPLE) {
+            this->textIdIndex = 11; // "Ah!"
+            this->showedCouplesMask = true;
+
+            if (this->npcEnMuto && this->npcEnBaisen) {
+                npcMuto->textIdIndex = 4;   // "Ah!"
+                npcBaisen->textIdIndex = 6; // "Ah!"
+            }
+        } else if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+            this->textIdIndex = 9; // "All must take refuge!!!"
+        }
+    }
+
+    this->actor.textId = sTextIds[this->textIdIndex];
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) && this->npcEnMuto != NULL && this->npcEnBaisen != NULL &&
+        (npcMuto->cutsceneState == 1 || npcBaisen->cutsceneState == 1)) {
+        EnDt_SetupMeetingCutscene(this, play);
+    }
+
+    // After completing Couple's Mask event and wearing Kafeis Mask
+    if (CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80) && (Player_GetMask(play) == PLAYER_MASK_KAFEIS_MASK)) {
+        this->actor.textId = 0x2368; // "My wife hired you? Go ask the Curiosity Shop"
+    }
+
+    Actor_OfferTalk(&this->actor, play, 150.0f);
+}
+
+void EnDt_SetupMeetingCutscene(EnDt* this, PlayState* play) {
+    EnMuto* npcMuto;
+    EnBaisen* npcBaisen;
+    s32 index;
+    s16* pTextData;
+
+    if (!CHECK_WEEKEVENTREG(WEEKEVENTREG_63_80)) {
+        if (this->npcEnMuto != NULL) {
+            if (this->npcEnBaisen != NULL) {
+                npcMuto = (EnMuto*)this->npcEnMuto;
+                npcBaisen = (EnBaisen*)this->npcEnBaisen;
+            }
+        }
+        if ((this->npcEnMuto != NULL) && (this->npcEnBaisen != NULL)) {
+            npcMuto->cutsceneState = 1;
+            npcBaisen->cutsceneState = 1;
+            if (Player_GetMask(play) == PLAYER_MASK_COUPLE) {
+                npcMuto->textIdIndex = 4;   // "Ah!"
+                npcBaisen->textIdIndex = 6; // "Ah!"
+                this->visualState = 5;
+                if (CHECK_WEEKEVENTREG(WEEKEVENTREG_60_08)) {
+                    this->visualState = 4;
+                }
+                EnDt_UpdateVisualState(this);
+            }
+        }
+    }
+
+    this->cutsceneState = EN_DT_CS_STATE_NONE;
+    for (index = 0; index < 24; index += 2) {
+        if ((play->msgCtx.currentTextId == CsIndexTable.D_80BEB268[index]) ||
+            (this->actor.textId == CsIndexTable.D_80BEB268[index])) {
+            this->cutsceneState = EN_DT_CS_STATE_WAITING;
+            this->csIdIndex = index;
+            break;
+        }
+    }
+
+    this->npcState = EN_DT_NPC_STATE_VIEWING_MEETING;
+    this->actionFunc = EnDt_UpdateMeetingCutscene;
+}
+
+void EnDt_UpdateMeetingCutscene(EnDt* this, PlayState* play) {
+    EnMuto* muto = NULL;
+    EnBaisen* baisen = NULL;
+    s32 index = CsIndexTable.D_80BEB26A[this->csIdIndex];
```

Comment:
```suggestion
    s32 index = sCsIndexTable[this->csIdIndex + 1];
```
The example of accessing D_80BEB26A I was talking about.